app/src/androidTest/java/nl/rug/www/rugsummerschools/controller/MainActivityTest.java

```diff
@@ -0,0 +1,169 @@
+import static android.support.test.espresso.matcher.ViewMatchers.assertThat;
+import static android.support.test.espresso.matcher.ViewMatchers.hasDescendant;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.anything;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+@RunWith(AndroidJUnit4.class)
+public class MainActivityTest {
+
+    private static final String DEFAULT_SCHOOL_NAME = "TestSchool";
+    private static final String DEFAULT_SCHOOL_ID = "5a5d15a9eff28d521340b136";
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityTestRule = new ActivityTestRule<>(MainActivity.class, true, false);
+
+    @Before
+    public void setUp() {
+        FirebaseAuth auth = FirebaseAuth.getInstance();
+        if (auth != null) {
+            auth.signOut();
+        }
+        LoginManager.getInstance().logOut();
+        List<LoginInfo> loginInfos = new ArrayList<>();
+        LoginInfo loginInfo = new LoginInfo();
+        loginInfo.setSchoolName(DEFAULT_SCHOOL_NAME);
+        loginInfo.setSchoolId(DEFAULT_SCHOOL_ID);
+        loginInfos.add(loginInfo);
+        ContentsLab.get().updateSchoolInfos(loginInfos);
+
+        mActivityTestRule.launchActivity(null);
+    }
+
+    @Test
+    public void allFragmentsDisplayed() {
+        onView(allOf(withId(R.id.section_name), withText(R.string.announcement))).check(matches(isDisplayed()));
+        onView(withId(R.id.main_view_pager)).perform(swipeLeft());
+        onView(allOf(withId(R.id.section_name), withText(R.string.general_info
+        ))).check(matches(isDisplayed()));
+        onView(withId(R.id.main_view_pager)).perform(swipeLeft());
+        onView(allOf(withId(R.id.section_name), withText(R.string.lecturer_info
+        ))).check(matches(isDisplayed()));
+        onView(withId(R.id.main_view_pager)).perform(swipeLeft());
+        onView(allOf(withId(R.id.section_name), withText(R.string.time_table
+        ))).check(matches(isDisplayed()));
+        onView(withId(R.id.main_view_pager)).perform(swipeLeft());
+    }
+
+    @Test
+    public void bottomSheetTest() {
+        onView(withId(R.id.navigation)).check(matches(hasDescendant(withText(R.string.notice))));
+        onView(withId(R.id.navigation)).check(matches(hasDescendant(withText(R.string.info))));
+        onView(withId(R.id.navigation)).check(matches(hasDescendant(withText(R.string.staff))));
+        onView(withId(R.id.navigation)).check(matches(hasDescendant(withText(R.string.schedule))));
+        onView(withId(R.id.navigation)).check(matches(hasDescendant(withText(R.string.forum))));
+    }
+
+    @Test
+    public void announcementTest() {
+        onView(withId(R.id.navigation_announcement)).perform(click());
```

Comment:
this is hard to read for somebody who is not familiar with the cod.e
but if for you it's clear, then it's ok.

Reply:
I thought it is quite a intuitive to understand this. I may add some comments what the test is for.